---
title: RIF and PMB combination
output: 
    md_document:
      preserve_yaml: FALSE
      fig_width: 7
      fig_height: 5
      toc: yes
      toc_depth: 2
    html_document:
      code_folding: hide
---

# Background 

We explored combination therapies of neglected and misused antibiotics. The
rifampin and polymyxin B combination emerged as a promising option against
various clinical isolates. This repository includes the data and code to
reproduce the analysis. For detailed figure captions and more information,
refer to the publication.


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo=FALSE, warning=FALSE, message=FALSE, dpi=300, fig.show = "hide")
knitr::opts_knit$set(global.par = TRUE)
library('data.table')
library('tidyverse')
library('DT')
library('cowplot')
library('BIGL')
library('drc')
library('rgl')

devtools::document('input/r')
suppressMessages(devtools::load_all('input/r'))

options(datatable.print.nrows=20)

theme_set(theme_html())
update_geom_defaults("point", list(size = 0.9))
update_geom_defaults("line", list(size = 0.5))
.text_size = 3
.point_size = 2
.facet_size = 8
theme_update(
  #legend.position = "bottom",
  strip.text.y = element_text(size = 8)
)
```

# Combinations

We studied the potential of the rifampicin-polymyxin B combination against
intra- and extracellular forms of bacteria: three _P. aeruginosa_ strains, 
two clinical isolates of _A. baumannii_, _E. coli, and _K. pneumoniae_.


```{r}
dat = loadDdiData('input/dat/rds/ddi_all_strains.rds')

# Response surface analysis
# rs = getRS(dat, 'rs')
# rs = readRDS('input/dat/tmp/rs.rds')
# 
# open3d()
# lapply(names(rs), sav3D)
```
![](output/fig/RsAllTheSpeciesStrains_portrait.svg)


# pH effect on synergy

```{r fig.cap='Acidification increases the number of drug concentration pairs with a synergistic effect. The number of significantly synergistic points are shown for the results above.'}
#getDdiCounts(readRDS('input/dat/tmp/rs.rds')) %>% 
#  group_by(Strain) %>% 
#  ggplot(aes(Cond, Syn)) +
#  geom_col(fill = 'grey70', col = 1) +
#  facet_wrap(~Name, nrow = 3, dir = 'v') +
#  labs(y = '# of synergistic concentrations', x = '') +
#  theme(strip.text.x = element_text(size = 12)) 
#ggsave(filename='output/fig/SynergyCountAcid.svg', bg='white', h=7.5, w=7.5)
```

![](output/fig/SynergyCountAcid.svg)


# pH effect on concentration-response

```{r SFig_monotherapy_dose-response_and_pH}
theme_set(theme_html())

# schematic for PD pars
pltPD = function() {
  tmp = data.table(
    #Conc      = c(0, 0.1, 0.3, 1, 3, 10, 30, 100),
    Conc      = c(0, 0.01, 0.03, 0.1, 0.3, 1, 3, 10)*10,
    Response  = c(3, 3, 2.8, 1.5, -0.5, -2.5, -2.8, -3.2)
  )
  
  tmp[, Response := Response + 1]
  
  # svg('output/fig/SFig_PDpar.svg', width = 3.5, height = 3.5)
  #oldpar = par()
  par(mar=c(3.1, 6.1, 0, 0.7), mgp = c(1.7, 0.5, 0), tcl=-0.2, pty='s')
  
  plot(
    (mod = drm(Response ~ Conc, data = tmp, 
        fct=LL2.4(names=c("h", "m", "b", "e"), fixed=c(1, NA, NA, NA)),
        lowerl = c(-2.5, NA, NA),
        robust = 'median'
    )),
      broken = T, pch = 19, 
      cex.axis=1.25, cex.lab=1.25,
      xt = c(0, 0.1, 1, 10),
      col = 'grey40',
      ylim = c(-2.5, 4.7), xlim = c(0, 50),
      xlab = c('Conc, mg/L'),
      ylab = expression(log[10]*'('*over('CFU'['24h'], 'CFU'['0h '])*')')
  )
  
  x1 = ED(mod, 0, type = 'absolute', logBase = exp(1), display = F)[1]
  x2 = ED(mod, 50, logBase = exp(1), display = F)[1]
  y2 = predict(mod, data.frame(Conc = x2))
  
  segments(x1, 0, x1, -100, lty = 2)
  segments(0.0001, 0, x1, 0, lty = 2)
  segments(x2, y2, x2, -100, lty = 2)
  segments(0.0001, y2, x2, y2, lty = 2)
  
  abline(h = coef(mod)[2], lty = 2)
  abline(h = coef(mod)[1], lty = 2)
  
  text(x1+3, y = 0 + 0.1, expression(C[s]))
  text(x2+3, y = y2 + 0.2, expression(EC[50]))
  text(x = 0.02, coef(mod)[2] + 0.3, expression(E[min]))
  text(x = 0.02, coef(mod)[1] + 0.4, expression(E[max]))
  # dev.off()
}

sub = subset(dat, strain == 'ATCC27853' & (d1 == 0 | d2 == 0))

p1 = function() {
  par(mar=c(3.1, 6.1, 0, 0.7), mgp = c(1.7, 0.5, 0), tcl=-0.2, pty='s')
  pltDR(effect ~ c1, subset(sub, c2==0), xlab = 'RIF, mg/L', zero=30, cex.legend=0.8, legendPos=c(1, -2), xt=c(0, 0.1, 10, 1000))
}
p2 = function() {
  par(mar=c(2.1, 5.1, 0, 0.7), mgp = c(1.7, 0.5, 0), tcl=-0.2, pty='s')
  pltDR(effect ~ c2, subset(sub, c1==0), col='orange', xlab = 'PMB, mg/L', cex.legend=0.8, legendPos=c(0.13, -2), xt=c(0, 0.01, 1, 100))
}

# pH
foo = fread("input/dat/raw/growth_and_pH.tsv")

foo = foo[time_h >= 0 & overnight_medium == 'pH7.4', .(
  od = gmu(od620),
  od.ci = gci(od620),
  ph = mean(pH),
  ph.ci = ci(pH)
  ), .(growth_medium, time_h)]

limit_of_detection = 0.03
foo[, od := replace(od, od < limit_of_detection, limit_of_detection)]

p0 = foo %>% 
  mutate(
    growth_medium = relevel(factor(growth_medium), 'pH7.4'),
    time_h = ifelse(time_h > 20, 14, time_h),
    NULL
    ) %>% 
  ggplot(aes(time_h, od, shape = growth_medium)) +
  scale_x_continuous(breaks = c(0, 5, 10, 14), labels = c(seq(0, 10, 5), 24), limits = c(0, 14)) +
  scale_shape_manual(name = '', values = c(19, 22, 15), labels = c('pH 7.4', 'pH 5.5',
      'pH 5.5 buffered')) +
  theme_html(12) +
  theme(legend.position = 'inside', 
    legend.position.inside = c(0.4, 0.9),
    aspect.ratio=1, 
    legend.text = element_text(size=11))

p3 = p0 + 
  scale_y_continuous(trans='log2', breaks = c(0.03, 0.3, 3)) + 
  geom_point(size=.point_size, col='grey20') +
  geom_pointrange(aes(ymin = od*od.ci, ymax = od/od.ci)) +
  geom_line(col='grey20') +
  coord_cartesian(ylim = c(0.03, 10)) +
  labs(x = 'Time, h', y = expression('OD'[620])) 

p4 = p0 %+% aes(y = ph) + 
  scale_y_continuous(position = 'right', limits=c(NA, 9)) +
  geom_point(size=.point_size, col=cbPalette[3]) +
  geom_line(col=cbPalette[3]) +
  geom_pointrange(aes(ymin = ph-ph.ci, ymax = ph + ph.ci), col=cbPalette[3]) +
  theme(legend.position = 'none',
    panel.border = element_blank(),
    axis.title.y.right=element_text(color = cbPalette[3]),
    axis.text.y.right=element_text(color = cbPalette[3]),
    axis.ticks.y.right=element_line(color = cbPalette[3]),
    axis.text.x = element_blank()
  ) +
  labs(x = '', y = 'pH')

foo = fread('input/dat/raw/charcoal.csv')
labs = c('RIF, mg/L', 'PMB, mg/L', 
  rep(expression(log[10]*'('*over('CFU'['24h'], 'CFU'['0h '])*')'), 2)
)

tmpFun = function (i, df, ...) {
  par(mar=c(3.1, 6.1, 0, 0.7), mgp = c(1.7, 0.5, 0), tcl=-0.2, pty='s')
  if(i == 1) bar = subset(df, ab == 'rif')
  else bar = subset(df, ab == 'pmb' | (ab == 'rif' & conc == 0))

  # fit char vs no-char but share Emin and EC50 (argument pmodels)
  fit = drm(response~conc, data=bar, CURVE, 
    pmodels=data.frame(CURVE,1,1),  
    fct=LL2.4( fixed=c(1, NA, NA, NA))) 
  
  plot(fit, type="average", cex.axis=1.25, cex.lab=1.25, lwd=2, 
    col='grey30', pch = c(1, 19), lty = 1, asp=1,
    xlab = labs[i], ylab = labs[i+2], broken=T, ...,
    ylim=c(-4,4), legendText=c("control", "charcoal")
  )
  plot(fit, type="confidence", add=T, legend=F, 
    col=c('grey70', 'grey10')
  )
}

p5 = function() tmpFun(1, foo, xt=c(0, 0.1, 10, 1000))
p6 = function() tmpFun(2, foo, xt=c(0, 0.01, 1, 100))

aligned = align_plots(p3, p4, align = 'hv', axis = 'tblr')

plot_grid(
  ggdraw(pltPD), ggdraw(p1), ggdraw(p2), 
    ggdraw(aligned[[1]]) + draw_plot(aligned[[2]]),
    ggdraw(p5), ggdraw(p6),
  labels='AUTO', label_size=16,
  align = 'hv', 
  nrow=2
)

# ggsave('output/fig/SFig_mth_dr_pH_charcoal_2.svg', width=10, height=6)
# ggsave('output/fig/SFig_mth_dr_pH_charcoal_2.pdf', width=10, height=6)
```

![](output/fig/SFig_mth_dr_pH_charcoal.svg)


# PD parameters

```{r}
# add Tn screen's PA14 (Liberati), and pmbR strains PA947 and PA1292 
foo = rbind(
  # pa14
  fread('input/dat/raw/pa14_mut_pmb_rif_cfu.csv')[mut == 0] %>% 
    .[, `:=` (genus   = 'Pseudomonas',
              strain  = 'PA14 Liberati',
              mut     = NULL)],
  # pmbR
  fread('input/dat/raw/pmb_resistant_isolates.csv') %>% 
    .[, genus := 'Pseudomonas']
  ) %>% 
  .[, `:=` (
    c1 = d1*mic1,
    c2 = d2*mic2)]

# transform
dat_fit = rbind(foo, dat, fill = T)[c1 == 0 | c2 == 0,
  ][, dose := ifelse(d1 == 0, c2, c1)
  ][, ab   := ifelse(d1 == 0, 2, 1)
  ][, .(data = list(.SD)), .(strain, cond)]

# fit four parameter logistic regression
dat_fit[, fit := map(data, ~drm(effect ~ dose, ab, data = .x, 
  fct=LL2.4(names=c("h", "m", "b", "e"), fixed=c(1, NA, NA, NA)),
  pmodels = data.frame(ab, 1, ab),
  lowerl = c(-4.5, NA, NA)))]

# get parameters
dat_fit[, pars  := map(fit, ~getPDPar(.x))]
dat_fit = dat_fit[
  dat_fit[, data[[1]][1, .(genus, mic1, mic2)], .(strain, cond)], 
  on = c('strain', 'cond')]

# reformat
dat_fit = unnest(dat_fit, pars) %>% as.data.table
dat_fit = getPdReadyForPlotting(dat_fit, dat)

p1 = dat_fit[!grepl('C', par)] %>% 
  .[lwr < -10, c('lwr', 'upr', 'value') := NA] %>% 
  ggplot(aes(value, strain, color = genus)) +
    geom_vline(xintercept = 0, col = 'white') +
    geom_pointrange(aes(xmin = lwr, xmax = upr), size = 1) +
    geom_text(aes(label = round(signif(value, 2), 1)), col = 1, size = 2.5) +
    facet_grid(cond ~ par, labeller = label_parsed, 
      scales = 'free', space = 'free_y') +
    geom_blank(aes(x = min)) +
    geom_blank(aes(x = max)) +
    scale_color_manual(values = cbPalette[4:1], guide = guide_legend(
        label.theme = element_text(angle = 0, face = "italic"), reverse = T)) +
    scale_fill_manual(values = cbPalette[4:1]) +
    labs(y = '', color = '') +
    theme_html() +
    theme(legend.position = 'top')

p2 = p1 %+%
  mutate(dat_fit[grepl('EC', par)], 
    value = ifelse(value > 512, 512, value), 
    mic = ifelse(mic > 512, 512, mic)) +
  geom_point(aes(mic, col = genus), size = 4.5, pch = 0) +
  geom_text(aes(x = mic, label = round(mic, 1)), size = 2.5) +
  scale_x_continuous(trans = 'log10', limits = c(0.03, 512), 
    labels = function(x) ifelse(x == 0, "0", x)) +
  facet_grid(cond ~ par, labeller = label_parsed, 
    scales = 'free', space = 'free_y') +
  xlab('Conc, mg/L') +
  theme(legend.position = 'none')

plot_grid(p1 + xlab(expression(log[10]*'('*over('CFU'['24h'], 'CFU'['0h '])*')')), 
  p2, rel_widths = c(3, 2), align = 'hv', axis = 'tb')

# ggsave('output/fig/SFig_mth_pd.svg', width=10, height=6)
```

![](output/fig/SFig_mth_pd.svg)


# Time-kill

```{r SFig_TimeKill}
dtk = fread('input/dat/raw/time-kill.csv') %>% 
  remNonCharcoal() %>% 
  addNewVariablesToTimeKill()

# pltTimeKill(dtk)
#ggsave('output/fig/SFig_TimeKill.svg', width=15, height=5)

# # Print out the concentrations used
# unique(dtk[time_h > 0], by = c('broth', 'group')) %>% 
#   .[, .(broth, group, d_1, d_2)] %>% 
#   .[order(group)]
```
![](output/fig/SFig_TimeKill.svg)

# Reverse Genetics Screen

To account for the synergy in molecular terms—beyond a nonspecific increase in membrane permeability by polymyxin B—we turned to chemical genetics ([Brochado and Typas, 2013](https://doi.org/10.1016/j.mib.2013.01.008)). Working with ordered PA14 transposon library ([Liberati et al., 2006](https://doi.org/10.1073/pnas.0511100103)), we derived a growth measure for monotherapies and combinations using colony opacity ([Kritikos et al., 2017](https://doi.org/10.1038/nmicrobiol.2017.14)). To account for plate-to-plate variation, the opacity was multiplicatively corrected. This results in zero-centering of the Bliss scores, which were derived next. The significance of difference from zero Bliss score, for any mutant, was estimated by a T-test (5 biological replicates) and corrected for multiple testing (Benjamini-Hochberg).

```{r include=F}
dat = loadIrisFiles("input/dat/raw/iris_files") %>% 
  addIrisVars() %>% 
  # Manual curation by visual inspection of images
  remBadPlates() %>% 
  flagBadColonies() %>% 
  turnBadColoniesToNAs() %>% 
  # Automatic curation by numbers
  setMisdetectedColoniesToNAs() %>%   # colonies missing from 60% of plates
  # plate-to-plate correction, brings plate opacities to same median
  correctOpacityForPlates()

# rep.correlations = getPairwiseReplicateCorrelations(dat)
# candidate.files.to.kick.out = getCandidateFilesToKickOut(rep.correlations)
# candidate.files.to.kick.out[ratio.discordant.vs.all==1, filename]
# [1] "LBpH5.5-Rifampicin16Polymyxin2-015.JPG.iris"

# Fitness ----------------------------------------
condition.vs.control = long2wide(dat)  # also average controls using median
#remember: opacity was replaced by the corrected opacity
condition.vs.control[,f.condition:=opacity/control.median]
# dim(condition.vs.control[is.na(f.condition)])[1] / dim(condition.vs.control)[1]
# [1] 0.02289894  #2.2% of fitness values are NA

# Epsilon ----------------------------------------
drug.merge = getEpsilons(condition.vs.control)

drug.merge = addAnnotation(drug.merge)
drug.merge = tTest(drug.merge)
chem.gen.res = getTTestResults(drug.merge) %>% 
  correctForMultipleTesting()
```

```{r overview_of_hits}
tmp = chem.gen.res %>% 
  mutate(t.test.q.value = ifelse(t.test.q.value < 0.001, 0.001,
      t.test.q.value)) %>% 
  mutate(showname = ifelse(
      (t.test.q.value < 0.05 & (gene.name != '' | gene.name.to.show %in% c('66480', '26590', '43270', '02150'))) | t.test.q.value < 0.01 ,
      T, F)) 

# tmp[t.test.q.value < 0.05 | t.test.q.value.gene < 0.05, .N, media]
# #      media  N
# # 1:      LB 65
# # 2: LBpH5.5 39

# # only 3 are shared between the media
# foo = tmp[media == 'LB' & (t.test.q.value < 0.05 | t.test.q.value.gene < 0.05), Tn.mutant.id] 
# bar = tmp[media == 'LBpH5.5' & (t.test.q.value < 0.05 | t.test.q.value.gene < 0.05), Tn.mutant.id] 
# tmp[Tn.mutant.id %in% intersect(foo, bar)]


getLocus = function (dat, condition) {
  unique(dat[eval(parse(text=condition))], by = c('locus'))[, locus] 
}

by_gene       = getLocus(tmp, 't.test.q.value.gene < 0.05')
by_mut        = getLocus(tmp, 't.test.q.value < 0.05')
by_both       = getLocus(tmp, 't.test.q.value < 0.05 | t.test.q.value.gene < 0.05')
by_gene_only  = setdiff(by_gene, by_mut)

# # 70 hits
# length(by_both)
# # 53 from mutants
# intersect(by_mut, by_both) %>% length()
# # 17 additional from genes
# setdiff(by_gene, by_mut) %>% length() 
# # 31 shared between mutants and genes 
# intersect(by_gene, by_mut) %>% length()
```

Of the `r length(by_both)` hits, `r length(intersect(by_mut, by_both))` are from mutant level of analysis and `r length(setdiff(by_gene, by_mut))` come from gene level of analysis. `r length(intersect(by_gene, by_mut))` are shared between mutants and genes i.e. hits regardless if mutant or gene level of analysis is used.



```{r ChemGen_LB}
(p0 = filter(tmp, media == 'LB') %>%  
  {
  ggplot(., aes(epsilon.median.mutant.condition, -log10(t.test.q.value), col = t.test.q.value < 0.05)) +
  geom_point(size = 2.5) +
  scale_color_manual(values = c('grey70', 'grey20')) +
  ggrepel::geom_text_repel(data= . %>% .[(showname)], 
    aes(label=gene.name.to.show), size = 6,
    max.overlaps = Inf, box.padding=0.75, seed=3) +
  lims(x = c(-0.8, 0.6), y = c(0, 3)) +
  labs(x = 'Shift in Bliss score', y = expression('log'[10]*'(p-value)')) +
  theme(aspect.ratio = 1, legend.position = 'none')

})
# ggsave('output/fig/ChemGen_LB.svg', width=5, height=5)
```

```{r ChemGen_LBpH5.5}
p0 %+% filter(tmp, media == 'LBpH5.5')
# ggsave('output/fig/ChemGen_LBpH5.5.svg', width=5, height=5)
```
|**LB**                        |**LB pH 5.5**             |
|:-----------------------------|:-------------------------|
![](output/fig/ChemGen_LB.svg) | ![](output/fig/ChemGen_LBpH5.5.svg)


```{r revgen-venn}
getGenes = function (dat) {
  out = list()
  for(i in unique(dat$media))
    out[[i]] = unique(dat[t.test.q.value < 0.05 & media == i], by='locus') %>%
      .[, (gene.name.to.show)]
  return(out)
}

foo = getGenes(chem.gen.res)
#vtable = gplots::venn(foo)
```

## GO terms

With the following, we bring some biological knowledge into the analysis. This
will get us at the level of processes/compartments as opposed to individual
genes. We will focus on LB beacuse we have more data from there which is also
more reliable (75% of the unique hits come from LB; there is less variance). In
addition, our results suggest, the effect of pH on synergy is weak in
PA14.


We did Gene Set Enrichment Analysis (GSEA) using GO terms from [pseudomonas.com
website](https://www.pseudomonas.com/) and Kologorov-Smirnov testing for
statistical significance estimation. Although most common approach, it has been
critizised for example
[here](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3134237/). 

### Cell component

```{r topGO}
library('topGO')
tab = getTopGO(chem.gen.res)

tmpFun = function (dat, name='CC') {
  dat[[name]][1:10, c('GO.ID', 'Term', 'raw.p.value')] %>% 
    mutate(Pvalue = round(as.numeric(raw.p.value), 4))  %>% 
    dplyr::select(-raw.p.value) %>% 
    mutate(`GO.ID` = sub('GO:', '', `GO.ID`)) 
}

tmpFun(tab) %>%
  knitr::kable()
  # knitr::kable(format='latex', booktabs=T) %>% 
  # kableExtra::save_kable('output/fig/table_cell_component.png', font='Verdana')
```

### Biological process

```{r}
# biological processes
tmpFun(tab, 'BP') %>% 
  knitr::kable()
  # knitr::kable(format='latex', booktabs=T) %>% 
  # kableExtra::save_kable('output/fig/table_biological_process.png', font='Verdana')
```

## Protein-protein interaction network

Protein-protein interaction (PPI) analysis using STRING database. There was no
data on PA14, so we will use PAO1 data to build and analyse the network onto
which we then map PA14 orthologs. 

```{r SFig_ppi_cluster}
library(igraph)
library(stringr)

hits = addAnnotation(chem.gen.res[t.test.q.value < 0.05, .(colony)])
hit_list = unique(hits[PAO1.ortholog != '', PAO1.ortholog])

# bring in the interactome
string_11_pa = 'input/dat/raw/pao1_ppi_stringDB_287.protein.links.v11.0.txt' %>% 
  fread() %>% 
  .[, `:=` (
    combined_score = combined_score/1000,
    #strip / trim species ID
    protein1 = gsub("287.DR97_", "PA", protein1, fixed=TRUE),
    protein2 = gsub("287.DR97_", "PA", protein2, fixed=TRUE)
  )]

# create an igraph object
pa_ppi = graph_from_data_frame(string_11_pa, directed = FALSE)
# summary(pa_ppi)
V(pa_ppi)$degree = degree(pa_ppi)

pa_ppi = simplify(pa_ppi, remove.multiple=T, remove.loops=T, edge.attr.comb='mean')

# extract network for all the hits in the provided list
hit_ppi = induced_subgraph(pa_ppi, which(V(pa_ppi)$name %in% hit_list))

# store degree on the network (main hit network)
V(hit_ppi)$degree = degree(hit_ppi)

# Replace the PAO1 genes with corresponding PA14 ones
lookup = unique(hits[, .(PAO1.ortholog, gene=gene.name.to.show)], by='PAO1.ortholog')
V(hit_ppi)$name = left_join(data.table(PAO1.ortholog=V(hit_ppi)$name), lookup)[, gene]

# Communities ------------------------------
# Community detection based on edge betweenness (Newman-Girvan)
# pdf('output/pdf/graph.pdf', height=10, width=20)

par(mfrow=c(1, 1), mar=c(1,3,1,1), cex=0.6)

set.seed(1)
ceb = cluster_edge_betweenness(hit_ppi) 
V(hit_ppi)$community = ceb$membership
cols = ins(cbPalette, 'white', c(5, 8:9, 11:14))

#svg('output/fig/SFig_ppi_cluster.svg', width=20, height=20)
#dendPlot(ceb, mode="hclust", palette = myCols, cex=1.5, cex.axis=1.5)
plot(hit_ppi, vertex.color=cols[V(hit_ppi)$community],
  vertex.label.cex=2, edge.width=4, vertex.label.family="Helvetica",
  vertex.label.dist=0,  vertex.size=16,
  vertex.frame.color='black', vertex.label.color='black'
)
#dev.off()
```
![](output/fig/SFig_ppi_cluster.svg)

The major graph communities, using (Newman-Girvan's edge betweenness):

1. The light blue nodes are central and seem to be enriched in regulatory genes.
2. The grey nodes seem to be a signal transduction from membrane to the
   regulatory genes (in light blue).
3. Orange and pink nodes are mostly related to metabolism; those in orange have
   more membrane related terms than the pink nodes. 




# Screen Validation

We validate the sensitivity of identified candidate mutants in low
throughput and in liquid LB medium at pH 7.4. Instead of factorial (i.e.
checkerboard), we use a fixed ratio design [Tallarida et al
1997](https://doi.org/10.1016/s0024-3205(97)01030-8).

```{r}
dat = fread('input/dat/raw/pa14_mut_pmb_rif.csv') %>% 
  addDrugRatio() %>% 
  addAUC() %>% 
  addFitness() %>% 
  remDissimilarExperimentalConditions() %>% 
  addPA14MutAnnotation()

dat[, dose:= ifelse(cond_f %in% c('PMB', 'combo'), d1, d2)]
dat_lst  = split(dat, dat$mut)
fit = lapply(dat_lst, fitPA14MutDR, curveid = cond_f)

# Loewe response surface ------------------------------
# rs = getPa14RS(dat)

# We'll use a snapshot of results from 2021-01-02
rsl = readRDS('input/dat/rds/2020-01-02_rs_pa14.rds') %>% lapply(., function(x) x$rsl)
foo = lapply(rsl, function(x) summary(x$maxR)$totals) %>% rbindlist(idcol='mut')
# Mutant numbering is potential source of confusion; hence I assign here
# manually: wild-type is coded as mut 0, fitting of mut 28 failed
foo[, mut := c(0:27, 29:45)]  
#arrange(foo, Syn) %>% print(20)
```

```{r}
pltNSynergiesOfValidatedMutants()
# ggsave('output/fig/validated_mutants_synergy_count.svg', w=5, h=10)
```

## Dose-response

```{r SFig_45PA14MutDoseResponses}
#svg('output/fig/SFig_45PA14MutDoseResponses.svg', 12, 16)
#plt45PA14MutantDR()  # dose-responses
#dev.off()
```
![](output/fig/SFig_45PA14MutDoseResponses.svg)


## Compare to Loewe's null

```{r SFig_45PA14MutCompLoeweNull}
#svg('output/fig/SFig_45PA14MutCompLoeweNull.svg', 12, 16)
# We'll use a snapshot of results from 2021-01-02
#plt45PA14MutantComparisonToLoeweNull('2020-01-02_rs_pa14')
#dev.off()
```

![](output/fig/SFig_45PA14MutCompLoeweNull.svg)


## Table with _E. coli_ orthologs

```{r}
getValidHitTable() %>% addEcOrthologs() %>% .[,-'locus_pa'] %>% 
  knitr::kable()
  # knitr::kable(format='latex', booktabs=T) %>% 
  # kableExtra::save_kable('output/fig/table_valid_hit.png')
```


## GO term table

```{r}
getGoTermTable() %>% 
  knitr::kable() 
  #knitr::kable(format='latex', booktabs=T) %>% 
  #kableExtra::save_kable('output/fig/table_valid_hit_go.png', font='Verdana')
```

# Selected five mutants

## 3D

```{r surface_analysis}
#rs = fread('input/dat/raw/pa14_mut_pmb_rif_cfu.csv') %>% 
#  prepForRS()  %>% 
#  setDT()  %>% 
#  addMutGeneNames() %>% 
#  getRS(., 'rs_6mut')
rs = readRDS('input/dat/tmp/rs_6mut.rds')
```

```{r}
rsl = lapply(rs, function(x) lapply(x, function(y) y$rsl))

# lapply(names(rs), sav3D, xcap=10)
```

![](output/fig/RsPA14Mutants_portrait.svg)

## ROS production & lipid peroxidation 

![](./input/fig/Fig5_ROS.png)


<!--
## Membrane permeability

```{r}
memb = readRDS('input/dat/rds/permeability.rds')
pltMemb(group_by(memb, PmbConcMgL, Strain))
#ggsave('./output/fig/PA14MembranePermeability.svg', w=15, h=5)
```

```{r}
memb = readRDS('input/dat/rds/permeability.rds')
om = addAovDunnetMembrane(memb) 
im = addAovDunnetMembrane(memb, membrane = 'IM') 
getDunnetResultsMemb(om, lev = 1)
getDunnetResultsMemb(im, lev = 1)
```
-->

## Supplementary Membrane Assays

![](output/fig/SFig_MembraneAssays.svg)

# PMB resistant isolates

```{r}
dat = 'input/dat/raw/pmb_resistant_isolates.csv' %>%  
  fread() %>% 
  setnames(., 'strain', 'mut') %>% 
  prepForRS(swapnames=T) %>% 
  setnames(., 'mut', 'strain') %>%
  setDT()

# rs = getRS(dat, 'rs_pmbR_intra')
# lapply(names(rs), sav3D)
```

![](output/fig/rsPMBr_intra.svg)


# Change in EC90 of RIF

```{r include=F}
dat = loadDdiData('input/dat/rds/ddi_all_strains.rds')

dat_fit = anti_join(dat, dat[c1==0 & c2!=0])[, .(data = list(.SD)), .(strain, cond)]

dat_fit[, fit := 
  map(data, function(x) {try(
    drm(effect ~ d1, d2, data = x, 
    fct=LL2.4(names=c("h", "m", "b", "e"), fixed=c(1, NA, NA, NA)),
    lowerl = c(-4.5, NA, NA))
  )})
]

idx = sapply(dat_fit$fit, function(x) class(x) != 'try-error')

dat_fit[idx, ed90  := map(fit, function(x) list(try(ED(x, 90, interval='fls', display=F) %>% 
      round(3))))]

dat_fit[idx, par90 := lapply(unlist(dat_fit[['ed90']], recursive=F), function(x) try(as.data.table(x, keep.rownames='level')))]

par90 = lapply(1:nrow(dat_fit[idx, ]), 
  function(x) dat_fit[idx, ][['par90']][[x]]) %>% 
  rbindlist(idcol='id') %>% 
  merge(dat_fit[, .(strain, cond)][idx][, id := 1:26], .) %>% 
  lowerCaseNamesDT() %>% 
  .[, level := gsub('e:|:90', '', level)] %>% 
  .[, effect:= 90]

set.seed(1)
res = par90 %>% 
  group_by(id) %>% 
  mutate(rel_IC = estimate/head(estimate, 1),
    rel_IC = ifelse(rel_IC > 2, 2, rel_IC)
    ) %>% 
  mutate(level = ifelse(level < 1 & level != 0, 0.1, level)) %>% 
  filter(as.numeric(level) <= 1) 

# res %>% group_by(level) %>% summarise(ec90=median(estimate), quantile(estimate, 0.2), ci=gci(estimate), lwr=ec90/ci, upr=ec90*ci) %>% dplyr::select(-ci)
#   level  ec90   lwr   upr
# 1 0      7.41 6.34   8.65
# 2 0.1    3.20 2.32   4.41
# 3 1      2.18 0.966  4.92
# res %>% filter(level==0.1) %>% pull(estimate) %>% quantile(0.2) 
#   20% 
# 0.939 

ggplot(res, aes(level, estimate)) + 
  geom_jitter(width=0.2, pch=21, aes(fill=cond), col='grey70') +
  geom_boxplot(alpha=0.5, outlier.shape=NA) +
  scale_y_continuous(name=expression('EC'[90]^'RIF'*', mg/L'), 
    trans='log2', breaks=c(16, 8, 4, 2, 1, 0.5, 0.25)) +
  scale_x_discrete(name='PMB, xMIC', labels=c('0', '< 1', '1')) +
  coord_cartesian(ylim = c(0.2, 16)) +
  labs(fill = '') +
  theme(aspect.ratio=1)

#ggsave('output/fig/rif_ec90_upon_pmb.svg', width=5, height=4, bg='white')
```

![](output/fig/rif_ec90_upon_pmb.svg)


<!--

# Chloroquine

```{r}
pacman::p_load(sandwich)
# https://stackoverflow.com/questions/65699987/coeftest-gives-error-with-drc-class-object
registerS3method("estfun", "drc", drc::estfun.drc)
registerS3method("bread", "drc", drc::bread.drc)

compPars = function(mod, p, ...) {
  as.data.table(compParm(mod, p, display=F, ...))[, .(Estimate, `p-value`)] %>% sapply(., signif, 2)

} 

chloroquine = fread('./input/dat/raw/chloroquine.csv')

m1_rif = drm(response ~ mgL, curveid=chloroquine,
  data=subset(chloroquine, strain=='ATCC27853' & drug=='RIF'), 
  fct=LL.4(fixed=c(1, NA, NA, NA), names=c('slope', 'Emax', 'Emin', 'IC50'))
)

m1_pmb = update(m1_rif, data=subset(chloroquine, drug=='PMB' & strain=='ATCC27853'))
m2_rif = update(m1_rif, curveid=NULL)
m2_pmb = update(m1_pmb, curveid=NULL)

anova(m1_rif, m2_rif)
anova(m1_pmb, m2_pmb)
AIC(m1_pmb, m2_pmb)
AIC(m1_rif, m2_rif)


# RIF ==============================

summary(m1_rif)
# Model fitted: Log-logistic (ED50 as parameter) (3 parms)
# 
# Parameter estimates:
# 
#                   Estimate Std. Error  t-value   p-value    
# Emax:chloroquine -1.901870   0.098459 -19.3164 < 2.2e-16 ***
# Emax:control     -2.281402   0.086839 -26.2717 < 2.2e-16 ***
# Emin:chloroquine  3.509196   0.089456  39.2283 < 2.2e-16 ***
# Emin:control      3.629471   0.105448  34.4194 < 2.2e-16 ***
# IC50:chloroquine 12.196531   1.252178   9.7403 1.722e-13 ***
# IC50:control      4.557944   0.451625  10.0923 4.932e-14 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error:
# 
#  0.2773383 (54 degrees of freedom)

compPars(m1_rif, 'IC50', vcov=sandwich)
# Estimate  p-value 
#  2.70000  0.00035 

compPars(m1_rif, 'Emax', vcov=sandwich)
# Estimate  p-value 
#  0.83000  0.00017 
compParm(m1_rif, 'Emax', vcov=sandwich)
# Comparison of parameter 'Emax' 
# 
#                     Estimate Std. Error t-value   p-value    
# chloroquine/control 0.833641   0.041126 -4.0451 0.0001675 ***
# CI95
ci = 0.833641 + c(1, -1) * 0.041126 * qt(0.025, df=54)
round(ci, 2)
# [1] 0.75 0.92


EDcomp(m1_rif, c(0, 0), type='absolute', vcov=sandwich)
# Estimated ratios of effect doses
# 
#                           Estimate Std. Error    t-value    p-value
# chloroquine/control:0/0 3.10350594 0.50911168 4.13171807 0.00012604

# CI95
ci = 3.10350594 + c(1, -1) * 0.50911168 * qt(0.025, df=54)
round(ci, 1)


ED(m1_rif, c(0), type='absolute', interval='delta', vcov=sandwich)
# Estimated effective doses
# 
#                 Estimate Std. Error    Lower    Upper
# e:chloroquine:0 22.50418    3.35270 15.78243 29.22593
# e:control:0      7.25121    0.49792  6.25295  8.24947

confint(m1_rif, 'Emax')
#                      2.5 %    97.5 %
# Emax:chloroquine -2.099268 -1.704472
# Emax:control     -2.455503 -2.107301



# PMB ==============================

summary(m1_pmb)
# Model fitted: Log-logistic (ED50 as parameter) (3 parms)
# 
# Parameter estimates:
# 
#                   Estimate Std. Error  t-value   p-value    
# Emax:chloroquine -0.640179   0.081292  -7.8750 1.570e-10 ***
# Emax:control     -1.094442   0.080318 -13.6263 < 2.2e-16 ***
# Emin:chloroquine  3.530741   0.090245  39.1240 < 2.2e-16 ***
# Emin:control      3.575761   0.088856  40.2421 < 2.2e-16 ***
# IC50:chloroquine  0.770553   0.097517   7.9017 1.421e-10 ***
# IC50:control      0.773965   0.083207   9.3017 8.315e-13 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error:
# 
#  0.2492302 (54 degrees of freedom)

compPars(m1_pmb, 'IC50', vcov=sandwich)
# Estimate  p-value 
#     1.00     0.98 


compPars(m1_pmb, 'Emax', vcov=sandwich)
# Estimate  p-value 
#  5.8e-01  1.4e-05 

compParm(m1_pmb, 'Emax', vcov=sandwich)
# Comparison of parameter 'Emax' 
# 
#                     Estimate Std. Error t-value   p-value    
# chloroquine/control 0.584936   0.087008 -4.7704 1.437e-05 ***
ci = 0.584936 + c(1, -1) * 0.087008 * qt(0.025, df=54)
round(ci, 1)
# [1] 0.4 0.8


EDcomp(m1_pmb, c(0, 0), type='absolute', vcov=sandwich)
# Estimated ratios of effect doses
# 
#                         Estimate Std. Error  t-value  p-value
# chloroquine/control:0/0 1.680623   0.259976 2.618017 0.011452
# CI95
ci = 1.680623 + c(1, -1) * 0.259976 * qt(0.025, df=54)
round(ci, 1)
# [1] 1.2 2.2

ED(m1_pmb, c(0), type='absolute', interval='delta', vcov=sandwich)
# Estimated effective doses
# 
#                 Estimate Std. Error   Lower   Upper
# e:chloroquine:0  4.24979    0.47379 3.29989 5.19968
# e:control:0      2.52870    0.27117 1.98503 3.07237

confint(m1_pmb, 'Emax')
#                       2.5 %     97.5 %
# Emax:chloroquine -0.8031596 -0.4771976
# Emax:control     -1.2554703 -0.9334134



```

```{r}
pacman::p_load(cowplot, ggpubr)

myTheme = theme(
  aspect.ratio = 1, 
  axis.line = element_blank(),
  axis.title = element_text(size=rel(1.1)),
  axis.text = element_text(size=rel(1.1)), 
  NULL
)

myplot = function(mod, ...) {
  dat = mod$data[, c(1:2, 4)]
  xrange = log(range(dat$mgL))
  xrange = seq(xrange[1], xrange[2], 0.001)
  fit = expand.grid(
    mgL = exp(xrange), 
    chloroquine = unique(dat$chloroquine)
  )
  fit$response = predict(mod, fit)

  dat %>%
    ggplot(aes(mgL, response, fill=chloroquine)) +
    geom_hline(yintercept=0, lty=2, col='grey30') +
    geom_line(data=fit, aes(col=chloroquine)) +
    geom_point(pch=21, col=1, size=2.5) +
    scale_x_continuous(trans='log10', 
      breaks=c(0.01, 0.1, 1, 10, 100, 1000), labels=function(x) as.character(x)) +
    scale_fill_manual(values=c('grey40', 'grey70'), name='') +
    scale_color_manual(values=c('grey40', 'grey70'), name='') +
    theme_cowplot() +
    myTheme + 
    panel_border(color=1) +
    ylab(expression(log[10]*'('*over('CFU'['24h'], 'CFU'['0h '])*')')) +
    ylim(-3, 4) +
    NULL
}

ggarrange(
  myplot(m1_rif) + xlab('RIF, mg/L'), 
  myplot(m1_pmb) + xlab('PMB, mg/L'), 
  legend = 'right',
  common.legend = T, 
  labels='AUTO'
)

ggsave('./output/fig/chloroquine.pdf', w=7.4, h=3.7)
```
-->

This repository is licensed under the MIT License - see the LICENSE file for details.
